tests/Fixture/Hooks/ObjectAfterInstantiateHook.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the zenstruck/foundry package.
+ *
+ * (c) Kevin Bond <kevinbond@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zenstruck\Foundry\Tests\Fixture\Hooks;
+
+use Zenstruck\Foundry\Hooks\AfterInstantiate;
+use Zenstruck\Foundry\Hooks\AsAfterInstantiateFoundryHook;
+use Zenstruck\Foundry\Tests\Fixture\Entity\EntityForHooks;
+
+#[AsAfterInstantiateFoundryHook(class: EntityForHooks::class)]
+final class ObjectAfterInstantiateHook
+{
+    /**
+     * @param AfterInstantiate<EntityForHooks> $hook
+     */
+    public function __invoke(AfterInstantiate $hook): void
+    {
+        $hook->object->name = "{$hook->object->name}\nObjectAfterInstantiateHook";
+    }
+}
```

Comment:
here is an example of a hook.

another syntax is possible:
```php
#[AsFoundryHook(event: AfterInstantiate::class(, class: EntityForHooks::class))]
final class AddressAfterInstantiateHook
{
    public function __invoke(AfterInstantiate $hook): void;
}
```
it would be clear then, that you should pass a `AfterInstantiate` to the `__invoke()` method.

I also wondered if we should not introduce an interface, but, that would mean 3 interfaces. And an attribute AND an interface seems redundant.

WDYT?

Reply:
I'd really prefer we didn't require the event argument - could be inferred from the type-hint.